Add new features

Andsbf opened the pull request:
add method to fetchs all contacts at 'once'
add user_info method  to fetch account holder information
add data_type legend

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -4,21 +4,106 @@ module API
+    # This data is from the API documentation
+    # [ Display_name, Label, Data_type ]
+    CONTACT_FIELDS = [
+      ['Address1 Type', 'Address1Type', 'string'],
```
not array

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -4,21 +4,106 @@ module API
+    USER_FIELDS = %w(
+      City Email EmailAddress2 EmailAddress3 FirstName HTMLSignature Id
+      LastName MiddleName Nickname Phone1 Phone1Ext Phone1Type Phone2
+       Phone2Ext Phone2Type PostalCode Signature SpouseName State
```
indentation

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -93,5 +182,41 @@ def find_by_field(table, limit, page, name, value, fields)
       connection('DataService.findByField', table, limit, page, name, value, fields)
     end
 
+    def user_info
```
```
def user_info(data = {})
end
```

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -93,5 +182,41 @@ def find_by_field(table, limit, page, name, value, fields)
 
+    def user_info
+      data = {}
+      query('User', 1000, page, data, USER_FIELDS)
```
where does `page` come from?

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -4,21 +4,106 @@ module API
+       StreetAddress1 StreetAddress2 Suffix Title ZipFour1
     )
 
+    DATATYPE_MAP = {'1' => 'Phone Number','2' => 'Social Security Number',
```
`CUSTOM_FIELD_TYPES` or similar

l1h3r replied:
maybe make it prettier

```
CONSTANT = {
  '1' => 'Phone Number',
  '2' => 'Social Security Number'
}
```

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -93,5 +182,41 @@ def find_by_field(table, limit, page, name, value, fields)
+
+      # This covers all the possible location of the tagID in the contact hash
+      datas =
+        [{ 'Groups' => tag_id }, { 'Groups' => "%,#{tag_id}" },
```
this can be better formatted

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -93,5 +182,41 @@ def find_by_field(table, limit, page, name, value, fields)
+    private
+
+    # Handles 1k+ contacs importing
+    def importer(data, fields)
```
`contact_importer`

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -93,5 +182,41 @@ def find_by_field(table, limit, page, name, value, fields)
+      imported = []
+
+      loop do
+        buffer = query('Contact', 1000, page, data, fields)
```
why `1000` ?

Andsbf replied:
1000 is the API max limit

l1h3r replied:
make it a constant

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -93,5 +182,41 @@ def find_by_field(table, limit, page, name, value, fields)
+      page =0
+      imported = []
+
+      loop do
```
no `loop`

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -93,5 +182,41 @@ def find_by_field(table, limit, page, name, value, fields)
+
+    # Handles 1k+ contacs importing
+    def importer(data, fields)
+      page =0
```
space

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -4,21 +4,123 @@ module API
+      '23' => 'Drilldown'
+    }
+    # This data is from the API documentation
+    # [ Display_name, Label, Data_type ]
```
needs to be updated to reflect the new structure

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -27,7 +129,11 @@ def get_tags(page = 0)
       data = { 'FormId' => -1 }
-      query('DataFormField', 1000, page, data, ['Name'])
+      response = query('DataFormField', 1000, page, data, ['Name', 'Label', 'DataType'])
+      response.map do |field|
```
`each`

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -4,21 +4,123 @@ module API
+      { label:'Website', name: 'Website', type: 'string'}
+    ]
+
+    CONTACT_FIELDS_LABELS = CONTACT_FIELDS.map { |fields| fields[:name] }
```
`CONTACT_FIELD_LABELS`

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -93,5 +199,28 @@ def find_by_field(table, limit, page, name, value, fields)
       connection('DataService.findByField', table, limit, page, name, value, fields)
     end
 
+    def user_info(data = {}, page = 0)
```
`get_user_info`

l1h3r commented on lib/infusionsoft/api.rb:
```diff
@@ -93,5 +199,28 @@ def find_by_field(table, limit, page, name, value, fields)
+        ]
+
+      datas.map do |data|
+        contacts += query('Contact', 1000, page, data, fields)
```
use `each`

Andsbf committed:
Add method to fetch contacts
gem version 0.1.2

Squashed commit of the following:

commit 3eb874ef1ca74d3a45e10e60ef35afe6ecc012c1
Author: Anderson Saunders <andsbf@gmail.com>
Date:   Thu Jul 9 13:35:10 2015 -0300

    Apply code review, make code cleaner

commit 71b3b662267ce8cb08701eb4ae6d244ee4a90c51
Author: Anderson Saunders <andsbf@gmail.com>
Date:   Wed Jul 8 20:48:56 2015 -0300

    make the code pretier

commit d08a88fd98a111405b2ba4dc455d3067b4a5ec15
Author: Anderson Saunders <andsbf@gmail.com>
Date:   Wed Jul 8 18:32:28 2015 -0300

    Add comment and more readable code

commit c60647c7e94800955ef1a15f89c621690fb51970
Author: Anderson Saunders <andsbf@gmail.com>
Date:   Wed Jul 8 18:30:36 2015 -0300

    fix bug in method return

commit a0dd12fe3431ce4eed5d54a67e6b688cdf40f31e
Author: Anderson Saunders <andsbf@gmail.com>
Date:   Wed Jul 8 17:14:56 2015 -0300

    Add method to fetch user(account holder) info, and fetch contacts at once without fetching ids first

update gem version
